File: app/src/test/java/eu/kanade/tachiyomi/injection/module/TestDataModule.java
package eu.kanade.tachiyomi.injection.module;

import android.app.Application;

import org.mockito.Mockito;

import eu.kanade.tachiyomi.data.database.DatabaseHelper;
import eu.kanade.tachiyomi.data.network.NetworkHelper;
import eu.kanade.tachiyomi.data.source.SourceManager;

public class TestDataModule extends DataModule {

    @Override
    DatabaseHelper provideDatabaseHelper(Application app) {
        return Mockito.mock(DatabaseHelper.class, Mockito.RETURNS_DEEP_STUBS);
    }

    @Override
    NetworkHelper provideNetworkHelper(Application app) {
        return Mockito.mock(NetworkHelper.class);
    }

    @Override
    SourceManager provideSourceManager(Application app) {
        return Mockito.mock(SourceManager.class, Mockito.RETURNS_DEEP_STUBS);
    }

}

File: app/src/test/java/eu/kanade/tachiyomi/injection/module/TestDataModule.kt
package eu.kanade.tachiyomi.injection.module

import android.app.Application
import eu.kanade.tachiyomi.data.network.NetworkHelper
import eu.kanade.tachiyomi.data.source.SourceManager
import org.mockito.Mockito

class TestDataModule : DataModule() {

    override fun provideNetworkHelper(app: Application): NetworkHelper {
        return Mockito.mock(NetworkHelper::class.java)
    }

    override fun provideSourceManager(app: Application): SourceManager {
        return Mockito.mock(SourceManager::class.java, Mockito.RETURNS_DEEP_STUBS)
    }

}
